spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeRegressor.java

```diff
@@ -0,0 +1,243 @@
+
+    @Description("The field on which to set the prediction. It will be of type double.")
+    private final String predictionField;
+
+    public DecisionTreeRegressorConfig(String fileSetName, String path, String featuresToInclude,
+                                       String featuresToExclude, String predictionField) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.featuresToInclude = featuresToInclude;
+      this.featuresToExclude = featuresToExclude;
+      this.predictionField = predictionField;
+    }
+
+    private void validate(Schema inputSchema) {
+      if (!Strings.isNullOrEmpty(featuresToExclude) && !Strings.isNullOrEmpty(featuresToInclude)) {
+        throw new IllegalArgumentException("Cannot specify values for both featuresToInclude and featuresToExclude. " +
+                                             "Please specify fields for one");
+      }
+      List<String> fields = getFeatureList(inputSchema);
+      for (String field : fields) {
+        Schema schema = inputSchema.getField(field).getSchema();
+        Schema.Type features = schema.isNullableSimple() ? schema.getNonNullable().getType() : schema.getType();
+        Preconditions.checkArgument(features.isSimpleType(), "Field to be used must be of type number: int, long, " +
+          "float, double but was of type %s for field %s ", features, field);
+        if (features.equals(Schema.Type.BYTES) || features.equals(Schema.Type.STRING) ||
+          features.equals(Schema.Type.BOOLEAN)) {
+          throw new IllegalArgumentException(String.format("Field to classify must be of type : int, double, " +
+                                                             "float, long but was of type %s for field %s ", features,
+                                                           field));
+        }
+        Preconditions.checkArgument(!features.equals(Schema.Type.NULL), "Field to classify must not be of type null");
+      }
+      Preconditions.checkArgument(inputSchema.getField(predictionField) == null, "Prediction field must not already " +
+        "exist in the input schema.");
+    }
+
+    private List<String> getFeatureList(Schema inputSchema) {
+      List<String> fields = new ArrayList<>();
+      List<Schema.Field> inputSchemaFields = inputSchema.getFields();
+      List<String> excludeFeatures = new ArrayList<>();
+      if (!Strings.isNullOrEmpty(featuresToInclude)) {
+        fields = Arrays.asList(featuresToInclude.split(","));
+        for (String field : fields) {
+          field = field.trim();
+          Schema.Field inputField = inputSchema.getField(field);
+          if (inputField == null) {
+            throw new IllegalArgumentException(String.format("Field %s does not exists in the input schema", field));
+          }
+        }
+      } else {
+        if (!Strings.isNullOrEmpty(featuresToExclude)) {
+          excludeFeatures = Arrays.asList(featuresToExclude.split(","));
```

Comment:
more robust to do `excludeFeatures.addAll(Splitter.on(',').trimResults().split(featuresToExclude))` since it will handle the scenario if a user includes whitespace before or after the comma.